chain/chain/src/flat_storage_resharder.rs

```diff
@@ -564,19 +563,24 @@ impl FlatStorageResharder {
                     );
                 }
             }
-            FlatStorageReshardingTaskResult::Failed
-            | FlatStorageReshardingTaskResult::Cancelled => {
-                // We got an error or a cancellation request.
+            FlatStorageReshardingTaskResult::Failed => {
                 // Reset parent.
                 store_update.set_flat_storage_status(
                     parent_shard,
                     FlatStorageStatus::Ready(FlatStorageReadyStatus { flat_head }),
                 );
-                // Remove children shards leftovers.
+                // Remove children shards entirely.
                 for child_shard in [left_child_shard, right_child_shard] {
                     store_update.remove_flat_storage(child_shard);
                 }
             }
+            FlatStorageReshardingTaskResult::Cancelled => {
+                // Remove children shards leftovers, but keep their state and the state of the
+                // parent as they are, so resharding can resume.
```

Comment:
Did you mean status instead of state? It seems like children's state is removed.

Reply:
Yeah good catch! I'll make this sentence clearer.